src/classes/Ingredient.js

```diff
@@ -1,11 +1,30 @@
 class Ingredient {
-  constructor(id, name, cost, amount, unit) {
-    this.id = id
-    this.name = name
-    this.estimatedCostInCents = cost
-    this.amount = amount
-    this.unit = unit
+  constructor(data) {
+    this.id = data.id
+    this.name = data.name
+    this.cost = data.cost
+    this.amount = data?.amount
+    this.unit = data?.unit
+  }
+
+  static fromIngredientData(id, ingredientsData, recipeData = null) {
+    const fromIngredients = ingredientsData.find(ingredient => id === ingredient.id)
+    let fromRecipe;
+    
+    let allIngredientData = {}
+
+    allIngredientData.id = id
+    allIngredientData.name = fromIngredients.name
+    allIngredientData.cost = fromIngredients.estimatedCostInCents
```

Comment:
Was this also being refactored to keep whole pantry ingredients? If it's a pantry ingredient, it would also need to have an `amount` set for it. Not sure if this was just for ingredients and recipe ingredients, or if we were transforming all ingredients (`ingredient`, `pantry ingredient`, `recipe ingredient` to be an instance of ingredient class.